Tests/TestUtils/XCTestCase+AnyCodableAsserts.swift

```diff
@@ -488,21 +488,34 @@ extension XCTestCase {
                 wildcardIndexes = Set(0..<expected.count)
                 expectedIndexes.removeAll()
             } else {
-                // TODO: update this to be flat? since there's only 1 operation now instead of 3
-                // Strongly validates index notation: "[123]"
+                // Finds wildcard indexes and extract their index values
+                
+                // Strongly validates array style access format: "[123]"
+                // Then filters all the path end keys that have valid array style access format, and returns only the string between the brackets
+                // Ex: "[123]" -> "123", "[ab12]" -> "ab12"
+                
+                // Wildcard index processing:
+                // 1. Filters the index values to the ones that have the wildcard marker `"*"`
+                // 2. Removes all asterisk characters from the index string
+                // 3. Casts the end result to an Int, discarding index if it is not a valid Int (sanitizes invalid strings)
+                // 4. Creates a unique set of the final values that pass all the above steps
+                
+                // Discard wildcard indexes that are out of bounds of the available expected indexes
+                // Then remove all wildcard indexes from the valid expected indexes, so validation is not performed twice
+                
                 let arrayIndexValueRegex = #"^\[(.*?)\]$"#
```

Comment:
nit: add comment
`// regex to extract indexes with wildcard matching identifier `*``